Source/WebCore/style/values/backgrounds/StyleBackgroundLayer.cpp

```diff
@@ -26,47 +26,51 @@
 #include "CachedImage.h"
 #include "StylePrimitiveKeyword+Logging.h"
 #include "StylePrimitiveNumericTypes+Logging.h"
-#include <wtf/PointerComparison.h>
-#include <wtf/TZoneMallocInlines.h>
 #include <wtf/text/TextStream.h>
 
 namespace WebCore {
 namespace Style {
 
+BackgroundLayer::BackgroundLayer()
+    : m_image(BackgroundLayer::initialImage())
+    , m_positionX(BackgroundLayer::initialPositionX())
+    , m_positionY(BackgroundLayer::initialPositionY())
+    , m_size(BackgroundLayer::initialSize())
+    , m_repeat(BackgroundLayer::initialRepeat())
+    , m_attachment(static_cast<unsigned>(BackgroundLayer::initialAttachment()))
+    , m_clip(static_cast<unsigned>(BackgroundLayer::initialClip()))
+    , m_origin(static_cast<unsigned>(BackgroundLayer::initialOrigin()))
+    , m_blendMode(static_cast<unsigned>(BackgroundLayer::initialBlendMode()))
+    , m_clipMax(static_cast<unsigned>(BackgroundLayer::initialClip()))
+{
+}
+
 BackgroundLayer::BackgroundLayer(ImageOrNone&& image)
-    : m_image(WTFMove(image))
-    , m_position({ BackgroundLayer::initialFillXPosition(), BackgroundLayer::initialFillYPosition() })
-    , m_size(BackgroundLayer::initialFillSize())
-    , m_repeat(BackgroundLayer::initialFillRepeat())
-    , m_attachment(static_cast<unsigned>(BackgroundLayer::initialFillAttachment()))
-    , m_clip(static_cast<unsigned>(BackgroundLayer::initialFillClip()))
-    , m_origin(static_cast<unsigned>(BackgroundLayer::initialFillOrigin()))
-    , m_blendMode(static_cast<unsigned>(BackgroundLayer::initialFillBlendMode()))
+    : BackgroundLayer { }
 {
+    setImage(WTFMove(image));
 }
 
-BackgroundLayer::BackgroundLayer(CSS::Keyword::None)
-    : BackgroundLayer { ImageOrNone { CSS::Keyword::None { } } }
+BackgroundLayer::BackgroundLayer(CSS::Keyword::None keyword)
+    : BackgroundLayer { }
 {
+    setImage(ImageOrNone { keyword });
 }
 
 BackgroundLayer::BackgroundLayer(RefPtr<StyleImage>&& image)
-    : BackgroundLayer { ImageOrNone { WTFMove(image) } }
+    : BackgroundLayer { }
 {
+    setImage(ImageOrNone { WTFMove(image) });
 }
```

Comment:
Should we do this instead?
```suggestion
BackgroundLayer::BackgroundLayer(RefPtr<StyleImage>&& image)
    : BackgroundLayer { ImageOrNone { WTFMove(image) } }
{
}
```